src/ui/features/onboarding/initialization.ts

```diff
@@ -0,0 +1,33 @@
+import { getCurrentUser } from 'src/shared/getCurrentUser';
+import { openOnboarding } from 'src/shared/openOnboarding';
+import { WindowContext, WindowParam } from 'src/ui/shared/WindowContext';
+import { OnboardingInterrupt } from './errors';
+
+async function getMode(windowContext: WindowContext) {
+  const hasOnboardingUrl = document.location.hash.startsWith('#/onboarding');
+  const hasExistingUser = Boolean(await getCurrentUser());
+  return hasOnboardingUrl || (!windowContext.isPopup() && !hasExistingUser)
+    ? 'onboarding'
+    : 'wallet';
+}
```

Comment:
I extracted this function. Yes, we access the browser storage _twice_ in the `maybeOpenOnboarding` function, but it's a bit easier to read this way